structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/service/CDPFlowStructuredEventHandler.java

```diff
@@ -0,0 +1,133 @@
+package com.sequenceiq.cloudbreak.structuredevent.service;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Scope;
+import org.springframework.messaging.Message;
+import org.springframework.statemachine.StateMachine;
+import org.springframework.statemachine.listener.StateMachineListenerAdapter;
+import org.springframework.statemachine.state.State;
+import org.springframework.statemachine.transition.Transition;
+import org.springframework.statemachine.trigger.Trigger;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.structuredevent.event.FlowDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.cdp.CDPStructuredEvent;
+import com.sequenceiq.flow.core.FlowEventListener;
+
+@Primary
+@Component
+@Scope("prototype")
+public class CDPFlowStructuredEventHandler<S, E> extends StateMachineListenerAdapter<S, E> implements FlowEventListener<S, E> {
```

Comment:
please add unit tests

Reply:
Hm, I tried to write any useful test, but I cannot. Could you please give me any suggestion?

Reply:
- in `public void transition(Transition<S, E> transition)` a `CDPStructuredEvent` is sent. in this meethod there are a lot of ternary operator and an if also. depending on the `transition` parameter one of the `cdpStructuredFlowEventFactory.createStructuredFlowEvent` is called with different values which could be asserted. 
you can even check the in case an error like nullpointer it is caught, so this would ensure no one removes the try catch by mistake.
- in `public void stateMachineStopped(StateMachine<S, E> stateMachine)` you could test it has only  effect if it's not complete. it also has some ternary operator, those effects could be tested also.

In the meanwhile: why do we have try-catch in `transition` and not in `stateMachineStopped`?

Reply:
_In the meanwhile: why do we have try-catch in transition and not in stateMachineStopped?_
Good questions, these codes are 2-3 years old.

Reply:
I don't agree with you. These tests could not catch any error. I cannot validate the content of the payload. It would be much useful than verifying the methods. 
But I will write these tests, maybe